tests/Model/BSONDocumentTest.php

```diff
@@ -23,6 +23,22 @@ public function testConstructorDefaultsToPropertyAccess(): void
         $this->assertSame('bar', $document->foo);
     }
 
+    public function testConstructorWithStandardObject(): void
+    {
+        $document = new BSONDocument((object) ['foo' => 'bar']);
+        $this->assertSame('bar', $document->foo);
+    }
+
+    public function testConstructorWithClassObject(): void
+    {
+        $document = new BSONDocument(new class () {
+            public string $foo = 'bar';
+            protected string $baz = 'qux';
+        });
+        $this->assertSame('bar', $document->foo);
+        $this->assertObjectNotHasProperty('baz', $document);
```

Comment:
This is essentially a test of ArrayObject itself. I think it'd be preferable to test `bsonSerialize()` (note: `jsonSerialize()` has the same implementation), so we can see what would actually get serialized to BSON.

My concern is that we should preserve the same encoding behavior as the extension, which will only encodes public properties.

See: https://3v4l.org/47VCv#v8.4.1

I think it's bit strange that the encoded names for protected and private properties still end up in output from `ArrayObject::getArrayCopy()`, and are preserved when casting back to an object. It's certainly odd that the stdClass instance resulting from that cast ends up with non-public properties, but that's likely an oddity of PHP itself.